### ISSUE TITLE
[bug]:  FreePBX17 - UCP daemon not running error

### ISSUE THREAD

**Issue by kapilgupta01** [FreePBX/issue-tracker#224]
### FreePBX Version

FreePBX 17

### Issue Description

Ref- https://community.freepbx.org/t/freepbx-17-ucp-interface/96943/10 

### Operating Environment

Freepbx 17

**Comment by microchipmatt** [comment]
After a clean installation of FreePBX 17 with no errors, UCP Daemon and Intrusion Protection (Fail2ban are running). After the restoration of the backup config with the "Backup & Restore" module, UCP Daemon, and Fail2ban are not running and will not start.

fwconsole pm2 --list
+---------------------------+-------+---------+----------+---------------------+------+----------+
| Process Name              | PID   | Status  | Restarts | Uptime              | CPU  | Mem      |
+---------------------------+-------+---------+----------+---------------------+------+----------+
| core-fastagi              | 11502 | online  | 0        | 7 hours, 33 minutes | 0.1% | 66.42MB  |
| sangomartapi              | 11509 | online  | 0        | 7 hours, 33 minutes | 0.1% | 106.95MB |
| core-calltransfer-monitor | 11512 | online  | 0        | 7 hours, 11 minutes | 0%   | 49.38MB  |
| ucp                       | 0     | errored | 15       | 0                   | 0%   | 0.00B    |
| restapps                  | 19194 | online  | 0        | 7 hours, 5 minutes  | 0%   | 55.84MB  |
+---------------------------+-------+---------+----------+---------------------+------+----------+

Node does not start with:
 fwconsole start ucpnode

Also, fail2ban will not start again until all log files for required daemons in jails are recreated. Unsure as to why loading the config deletes those logs.

**Comment by ramarajan222** [comment]
HI @microchipmatt 

I faced the same issue on my testing system. After enabling precedence ::ffff:0:0/96 100 in /etc/gai.conf and rebooting the system, the UCP restart issue was fixed.

If you haven't restarted the system, check after the restart. thanks

**Comment by kapilgupta01** [comment]
you can also try to "ping localhost" and see which ip its resolving, it should resolve to 127.0.0.1 not ipv6

**Comment by microchipmatt** [comment]
@ramarajan222 I did try  enabling this line yesturday and it did not resolve the issue, but I can try again. 
Current output of pinging localhost:
ping localhost
PING localhost (127.0.0.1) 56(84) bytes of data.
64 bytes from localhost (127.0.0.1): icmp_seq=1 ttl=64 time=0.021 ms
64 bytes from localhost (127.0.0.1): icmp_seq=2 ttl=64 time=0.018 ms
64 bytes from localhost (127.0.0.1): icmp_seq=3 ttl=64 time=0.015 ms
64 bytes from localhost (127.0.0.1): icmp_seq=4 ttl=64 time=0.016 ms
64 bytes from localhost (127.0.0.1): icmp_seq=5 ttl=64 time=0.016 ms

**Comment by microchipmatt** [comment]
It is already uncommented in my /etc/gai.conf:

#precedence  ::1/128       50
#precedence  ::/0          40
#precedence  2002::/16     30
#precedence ::/96          20
#precedence ::ffff:0:0/96  10
#
#    For sites which prefer IPv4 connections change the last line to
#
precedence ::ffff:0:0/96  100

**Comment by microchipmatt** [comment]
It is uncommented in my  /etc/gai.conf already:

![image](https://github.com/FreePBX/issue-tracker/assets/11633632/de9d9c7f-7f6a-4fbf-add9-087fc612e0d6)

UCP Node still does not start after reboot.
I can provide more log output but I would need to get fail2ban running first since this bug seems to destroy many of the logs when I restore my config.

**Comment by kapilgupta01** [comment]
hi @microchipmatt is this fresh system , i mean do you have any configuration done on this system ?
if not then can you please - 
1) apt-get install ssh-keys (from root user)
2) share the ip via email to kgupta@sangoma.com so we can ssh to your system to check what is going on.  do not share on this publicly accessible platform. thanks

**Comment by ramarajan222** [comment]
Hi @microchipmatt Have you checked the ucp status after the system restart?

**Comment by microchipmatt** [comment]
Yes, it was a fresh install. UCP was running, but when I restored my config from the backup,  UCP stopped working, and so did fail2ban.  I can roll back to the point of the Fresh FreePBX17 install on the VM, I snapshotted the machine. Since this is a test machine, for you to Access via SSH I would need to CUT you a port on my Public facing firewall and map you with a NAT to the machine giving only your IP access. MIght be better if we do some sort of remote session, I will contact you via email.

@ramarajan222 I did reboot after the change, but still no life.

**Comment by microchipmatt** [comment]
@kguptasangoma looking forward to connecting with you tonight if you have time. As another test, I do have chan_sip extensions still, but I let the restore convert them to pjsip in FreePBX17. Thinking that was possibly the issue, I reinstalled FreePBX 17. I downgraded asterisk to 20 and then restored my config again, but upon restoration on FreePBX17, fail2ban and UCP node will not start and are destroyed exactly as before. If I restore to FreePBX 16 on CentOS, it restores perfectly, and UCP and fail2ban are running…So It's got to be something with how Backup & Restore is restoring it on FreePBX 17. Looking forward to connecting with you tonight MST morning IST your time :)

If I restore the Backup to **FreePBX 16** on Cent OS, Everything works perfectly after restore UCP and Fail2 ban are running:
![image](https://github.com/FreePBX/issue-tracker/assets/11633632/ff77245f-0f4f-4557-8bf8-b2c550250120)

If I restore the same backup to **FreePBX 17** on Debian, UCP and fail2ban are destroyed:
![image](https://github.com/FreePBX/issue-tracker/assets/11633632/98863029-f145-4521-8b62-3cc0f7c8e63b)

**Comment by microchipmatt** [comment]
I have tried many different ways to restore my current config to FreePBX 17, and everytime, UCP and fail2ban breaks, as stated above. I can restore to FreePBX 16 on CentOS no problem, and UCP is not broken. When I restore my full backup config from backup & restore to FreePBX 17 on debian 12.5 UCP breaks. I cannot find a way to have UCP running after the restore. I have tried several methods and have not had a successful one with UCP running to date. Everthing else seems mostly intact, but UCP is broken.  Things I have tried:

- Restore to FreePBX 16 on CentOS which has UCP running after I am done. Take a backup of the intact FreePBX16 restore and restore to FreePBX17 on Debian [FAIL - UCP Daemon is Nuked]

- Take a backup of the intact FreePBX16 restore, leave out UCP in the hopes that the pre-running UCP on FreePBX 17 will stay intact once restored, in the hopes UCP will be running after restore is done [FAIL - UCP Daemon is Nuked]

- Take a Backup of the UCP module running on FreePBX 17 before Restoring from FreePBX 16 on CentOS which has UCP running after I am done. Take a backup of the intact FreePBX16 restore and restore to FreePBX17 on Debian, hoping I can restore UCP by itself afterward [FAIL - UCP Daemon is Nuked]. When  the backed up UCP module is retored it works, but ALL the extensions are nunked, along with other items.

Even though I can restore to FreePBX 16 with NO issues and have everything running including UCP, I cannot find a path to do the same within FreePBX 17, I believe there is definatly an issue here, that will affect many user's once they onboard, if this is not fixed.
